### PR TITLE
Blog app: add API endpoints

### PR DESCRIPTION
**Pull request by Estete9**
## Project requirements

NOTE: If you transitioned from 1.0 Microverse program use the [Social Media website](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/) you have built.

- Create an API endpoint to list all posts for a user.
- Create an API endpoint to list all comments for a user's post.
- Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should not be set by passing a parameter.
- Your API endpoints should receive JSON and respond JSON as well.

Need a big picture?
Remind me about [the big picture of this project](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md).

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -2,6 +2,10 @@ class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
     @posts = @user.posts.includes(:author, comments: [:user])
+    respond_to do |format|
+      format.html
+      format.json { render json: @posts }
+    end
```
Comment (shella12) [inline app/controllers/posts_controller.rb]: - [x] Good start with JSON responses: You've correctly implemented JSON rendering for some actions, which is essential for API responses. However, we need to focus on establishing dedicated API endpoints to create a well-structured and maintainable API.

### 1. Specific Points:

**Missing API Routes:**

- Current routes: Your routes are currently designed for HTML-based views, not separate API endpoints.

- Solution: Introduce a namespace block for API routes for example:

```rb
Rails.application.routes.draw do
  # ... other routes

  namespace :api, defaults: { format: :json } do
    resources :posts, only: [:index, :show] # Define API routes for posts
    # Add other API routes as needed
  end
end

```

### 2. Controller Actions:

Separate actions: Consider creating distinct controller actions for API responses, ensuring a clear separation of concerns.

Example:

```rb

class Api::PostsController < ApplicationController
  def index
    @posts = Post.all
    render json: @posts
  end

  # ... other API-specific actions
end


```

### 3.Additional Considerations [OPTIONAL] :

Versioning: For future API changes, consider versioning your API using namespaces like api/v1.
Authentication: If authentication is required, implement appropriate mechanisms for API access.

### 4. Testing Endpoints:

Importance: Use tools like Postman or curl to test your API endpoints and ensure they function as expected.

**Review by shella12 — CHANGES_REQUESTED** [review]
Hi <STUDENT_MENTION>, 

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- Added JSON responses ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (Estete9) [reply app/controllers/posts_controller.rb:8]: Hi @shella12 and thank you for your observation. I texted you in Slack, can you help me answering?

**Review by fobadara — APPROVED** [review]
Hi team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

APPRVED
## Highlights
✅ Descriptive README
✅ Descriptive PR summary
✅ Clean code
✅ Use of `/api` and versioning in routes


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**